### ISSUE TITLE
Code Boxes form collapses in  Knative Documentation 

### ISSUE THREAD

**Issue by Enoch-Kim** [knative/docs#2706]
## Expected Behavior
codes are exist in code box correctly

## Actual Behavior
it's come outside of boxes..
for example, it looks well in github 
(picture = https://drive.google.com/file/d/1szSe2UaqiHD_QAkK844YWnJy_GzhuKlR/view?usp=sharing
 link = https://knative.dev/docs/eventing/samples/kubernetes-event-source/)
but collapsed in documentation 
(picture = https://drive.google.com/file/d/1IIgfDdIJ-X5v_hlbzzjCoB9afAPFE_Xr/view?usp=sharing
 link = https://github.com/knative/docs/tree/master/docs/eventing/samples/kubernetes-event-source)

## Steps to Reproduce the Problem

1.
1.
1.

## Additional Info
**Additional context**

**Install information:**
 - Platform (GKE, IKS, AKS, etc.): Chrome Browser
 - Knative Version:

**Comment by evankanderson** [comment]
Here are the images (extracted from Google Drive):

It looks like the `---` separator between YAML sections is being interpreted as an `<hr>` by Hugo.

<img width="1038" alt="스크린샷 2020-07-29 오전 1 48 20" src="https://user-images.githubusercontent.com/7959095/88831016-993c2c80-d183-11ea-8a7b-f5511942825e.png">
<img width="962" alt="스크린샷 2020-07-29 오전 1 48 29" src="https://user-images.githubusercontent.com/7959095/88831029-9b9e8680-d183-11ea-8a02-e252f60de007.png">

**Comment by evankanderson** [comment]
Here's the generated HTML:

```html
<div class="highlight"><pre style="background-color:#f8f8f8;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-yaml" data-lang="yaml">apiVersion<span style="color:#000;font-weight:bold">:</span><span style="color:#f8f8f8;text-decoration:underline"> </span>v1<span style="color:#f8f8f8;text-decoration:underline">
</span><span style="color:#f8f8f8;text-decoration:underline"></span>kind<span style="color:#000;font-weight:bold">:</span><span style="color:#f8f8f8;text-decoration:underline"> </span>ServiceAccount<span style="color:#f8f8f8;text-decoration:underline">
</span><span style="color:#f8f8f8;text-decoration:underline"></span>metadata<span style="color:#000;font-weight:bold">:</span><span style="color:#f8f8f8;text-decoration:underline">
</span><span style="color:#f8f8f8;text-decoration:underline">  </span>name<span style="color:#000;font-weight:bold">:</span><span style="color:#f8f8f8;text-decoration:underline"> </span>events-sa<span style="color:#f8f8f8;text-decoration:underline">
</span><span style="color:#f8f8f8;text-decoration:underline">  </span>namespace<span style="color:#000;font-weight:bold">:</span><span style="color:#f8f8f8;text-decoration:underline"> </span>default<span style="color:#f8f8f8;text-decoration:underline">
</span><span style="color:#f8f8f8;text-decoration:underline">
</span></code></pre></div></li>
</ol>
<hr>
<p>apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
name: event-watcher
rules:
- apiGroups:
- &quot;&rdquo;
resources:
- events
verbs:
- get
- list
- watch</p>
<hr>
<p>apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
name: k8s-ra-event-watcher
roleRef:
apiGroup: rbac.authorization.k8s.io
kind: ClusterRole
name: event-watcher
subjects:
- kind: ServiceAccount
name: events-sa
namespace: default</p>
<pre><code>
 ```

**Comment by evankanderson** [comment]
This is https://github.com/yuin/goldmark/issues/111, which looks to be fixed in Hugo 0.72

I'm going to try an upgrade to the latest Hugo now, to see how that improves things.
